semanticsearch/workload.py

```diff
@@ -0,0 +1,9 @@
+def put_settings(es, params):
+    es.cluster.put_settings(body=params["body"])
+
+
+def register(registry):
```

Comment:
Do we need this? i see an operation is defined at opensearch-benchmark as "put-settings"

Reply:
without this I'm getting error when running my workload:

```
opensearch-benchmark execute-test --workload-path="/Users/user/dev/opensearch/opensearch-benchmark-workloads/semanticsearch/workload.json" --pipeline=benchmark-only  --target-host=http://localhost:9200 --kill-running-processes --workload-params="number_of_shards:6,max_num_segments:8,concurrent_segment_search_enabled:'false',ingest_percentage:1" --test-procedure=hybrid-query-aggs-full

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[ERROR] Cannot execute-test. Error in test execution orchestrator (Could not register workload plugin at [/Users/user/dev/opensearch/opensearch-benchmark-workloads/semanticsearch])
```

if there is a way to avoid it I would be happy to know

Reply:
Let me take a look and update here. I see that you are registering only if there is an import error. I am not sure why you have to do it only if there is an import error.

Reply:
@martin-gaievski The below command runs without any framework error. Can you check were you able to run below command?
```
opensearch-benchmark execute-test --workload-repository ~/WORK/opensearch-benchmark-workloads  --workload="noaa_semantic_search" --pipeline=benchmark-only  --target-host=http://localhost:9200 --kill-running-processes --workload-params="number_of_shards:6,max_num_segments:8,concurrent_segment_search_enabled:'false',ingest_percentage:1" --test-procedure=hybrid-query-aggs-full

```

Reply:
When it fails the `workload.py` file was there with the following content:

```
def put_settings(es, params):
    es.cluster.put_settings(body=params["body"])
```

This is the log file after this run: [benchmark.log](https://github.com/opensearch-project/opensearch-benchmark-workloads/files/15396875/benchmark.log)

when I add back the `register` function (same content as in this PR) it works again.

```
def put_settings(es, params):
    es.cluster.put_settings(body=params["body"])


def register(registry):
    try:
        from osbenchmark.worker_coordinator.runner import PutSettings
    except ImportError:
        registry.register_runner("put-settings", put_settings)
```

Reply:
> When it fails the `workload.py` file was there with the following content:
> 
> ```
> def put_settings(es, params):
>     es.cluster.put_settings(body=params["body"])
> ```
> 

Why do you need above method?

Reply:
I was able to run this workload without `workload.py`. Command that worked for me it following (one that you've shared earlier didn't work as it tries to checkout main branch that doesn't have my changes):

```
opensearch-benchmark execute-test --workload-path="/Users/user/dev/opensearch/opensearch-benchmark-workloads/noaa_semantic_search" --pipeline=benchmark-only  --target-host=http://localhost:9200 --kill-running-processes --workload-params="number_of_shards:6,max_num_segments:8,concurrent_segment_search_enabled:'false',ingest_percentage:1" --test-procedure=hybrid-query-aggs-full
```